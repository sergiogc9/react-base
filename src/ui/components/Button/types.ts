import { ButtonProps } from '@material-ui/core/Button';

export type ComponentProps = ButtonProps & {
   /**
   * The text color to use
   */
   textColor?: string
};
