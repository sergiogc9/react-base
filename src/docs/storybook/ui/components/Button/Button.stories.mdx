import { ArgsTable, Meta, Story, Canvas } from "@storybook/addon-docs/blocks";

import Button from "../../../../../ui/components/Button";
import { ComponentProps } from "../../../../../ui/components/Button/types";

<Meta
	title="UI Kit/Button"
	component={Button}
	argTypes={{
		textColor: { control: 'color' },
		onClick: { action: 'clicked' }
	}}
/>

# Button

Button component based on [Material-UI button](https://material-ui.com/components/buttons/).

## Sizes

<Canvas>
  <Story name="size-small" args={{size: 'small', children: 'Small button'}}>{args => <Button {...args} />}</Story>
  <Story name="size-medium" args={{size: 'medium', children: 'Medium button'}}>{args => <Button {...args} />}</Story>
  <Story name="size-large" args={{size: 'large', children: 'Large button'}}>{args => <Button {...args} />}</Story>
</Canvas>

## Types

<Canvas>
  <Story name="type-text" args={{variant: 'text', children: 'Text button'}}>{args => <Button {...args} />}</Story>
  <Story name="type-outlined" args={{variant: 'outlined', children: 'Outlined button'}}>{args => <Button {...args} />}</Story>
  <Story name="type-contained" args={{variant: 'contained', children: 'Contained button'}}>{args => <Button {...args} />}</Story>
</Canvas>

<ArgsTable of={Button} />
