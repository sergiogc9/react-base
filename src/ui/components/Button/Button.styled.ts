import React from 'react';
import styled, {css} from 'styled-components';
import Button from '@material-ui/core/Button';

import { ComponentProps } from './types';

const StyledButton = styled<React.FC<ComponentProps>>(Button)`
	&&{
		${props => props.textColor && css`color: ${props.textColor};`}
	}
`;

export default StyledButton;
