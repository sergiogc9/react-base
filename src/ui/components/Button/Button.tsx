import React from 'react';
import StyledButton from './Button.styled';

import { ComponentProps } from './types';

const Button: React.FC<ComponentProps> = props => {
	return <StyledButton {...props} color='inherit' />;
};

export default Button;
