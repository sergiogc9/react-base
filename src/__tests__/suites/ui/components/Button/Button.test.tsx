import React from "react";
import { render } from '@testing-library/react';

import Button from 'ui/components/Button';

describe('Button', () => {

	it("should render text", () => {
		const { baseElement } = render(<Button>This is a test button!</Button>);
		expect(baseElement.innerHTML).toMatchSnapshot();
	});
});
