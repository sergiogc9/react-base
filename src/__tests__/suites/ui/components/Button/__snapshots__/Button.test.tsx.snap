// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render text 1`] = `"<div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text sc-bdnylx hpzcRJ MuiButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">This is a test button!</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>"`;
